File: pythonic/List/List.py
from abc import ABC, abstractmethod

class List(ABC):

    @abstractmethod
    def clear(self):
        pass
    @abstractmethod
    def insert(self, item):
        pass
    @abstractmethod
    def append(self, item):
        pass
    @abstractmethod
    def remove(self):
        pass
    @abstractmethod
    def moveToStart(self):
        pass
    @abstractmethod
    def moveToEnd(self):
        pass
    @abstractmethod
    def prev(self):
        pass
    @abstractmethod
    def next(self):
        pass
    @abstractmethod
    def length(self):
        pass
    @abstractmethod
    def currPos(self):
        pass
    @abstractmethod
    def moveToPos(self, pos):
        pass
    @abstractmethod
    def getValue(self):
        pass
    @abstractmethod
    def Search(self, item):
        pass

File: List/cplusplus/List.cpp
template <class E>
class List
{
public:
    List() {}
    virtual ~List() {}
    virtual void clear() = 0;
    virtual void insert(E item) = 0;
    virtual void append(E item) = 0;
    virtual E remove() = 0;
    virtual void moveToStart() = 0;
    virtual void moveToEnd() = 0;
    virtual void prev() = 0;
    virtual void next() = 0;
    virtual int length() const = 0;
    virtual int curPos() const = 0;
    virtual void moveToPos(int pos) = 0;
    virtual E getValue() const = 0;
    virtual int Search(E item) const = 0;
};